File: Python/0005. N进制小数.py
# If you need to import additional packages or classes, please import here.

def func():
    # please define the python3 input here. 
    # For example: a,b = map(int, input().strip().split())
    # please finish the function body here.
    # please define the python3 output here. For example: print().
    while True:
        try:
            m,n = map(float,input().strip().split())
            if m == 0 and n == 0:
                return

            dig = 10
            num = 0
            power = 0.1

            while dig:
                num += power*int(m*n)
                m = m*n - int(m*n)
                dig -= 1
                power *= 0.1

            print("%.10f" %(num))
            
        except EOFError:
            break

    
if __name__ == "__main__":
    func()

File: C++/0005. N进制小数.cpp
// we have defined the necessary header files here for this problem.
// If additional header files are needed in your program, please import here.

#include <math.h>

int main()
{  
  // please define the C++ input here. For example: int a,b; cin>>a>>b;;  
  // please finish the function body here.  
  // please define the C++ output here. For example:cout<<____<<endl; 
    double m,n;
    while(cin>>m>>n){
        if (m == 0 && n == 0){
            break;
        }
        string result = "0.";
        cout<<result;
        int digit;
        for(int i=0;i<10;++i){
            digit = floor(m*n);
            m = m*n-digit;
            cout<<digit;
        }
        cout<<endl;
    }
    return 0;
}
